logicExpert.py

```diff
@@ -173,252 +37,264 @@ def updateLogic (unusedLocations, locArray, loadout) :
-            thisLoc['inlogic'] = vulnar and canUseBombs and (Speedball in loadout) and (SpeedBooster in loadout) and (GravitySuit in loadout) and ((DarkVisor in loadout) or (pirateLab and canUsePB))
-        if thisLoc['fullitemname'] == "Restricted Area":
-            thisLoc['inlogic'] = vulnar and canUseBombs and underwater and (MetroidSuit in loadout) and (((DarkVisor in loadout) and (Wave in loadout)) or ((pirateLab or eastLomyr) and canUsePB))
-        if thisLoc['fullitemname'] == "Foundry":
-            thisLoc['inlogic'] = vulnar and canUseBombs and underwater and (((DarkVisor in loadout) and (Wave in loadout)) or ((pirateLab or eastLomyr) and canUsePB))
-        if thisLoc['fullitemname'] == "Norak Escarpment":
-            thisLoc['inlogic'] = eastLomyr and (canFly or (SpeedBooster in loadout))
-        if thisLoc['fullitemname'] == "Glacier's Reach":
-            thisLoc['inlogic'] = upperVulnar and (energyCount > 3)
-        if thisLoc['fullitemname'] == "Sitting Room":
-            thisLoc['inlogic'] = upperVulnar and canUsePB
-        if thisLoc['fullitemname'] == "Suzi Ruins Map Station Access":
-            thisLoc['inlogic'] = suzi
-        if thisLoc['fullitemname'] == "Obscured Vestibule": 
-            thisLoc['inlogic'] = suzi
-        if thisLoc['fullitemname'] == "Docking Port 3": # (3 = letter Gamma)
-            thisLoc['inlogic'] = (((spaceDrop in loadout) == False) and (Grapple in loadout)) or ((spaceDrop in loadout) and geothermal and (Grapple in loadout) and (Screw in loadout) and (MetroidSuit in loadout))
-        if thisLoc['fullitemname'] == "Arena":
-            thisLoc['inlogic'] = vulnar and (Morph in loadout)
-        if thisLoc['fullitemname'] == "West Spore Field":
-            thisLoc['inlogic'] = vulnar and canUseBombs and (Super in loadout) and (Wave in loadout) and (Speedball in loadout)
-        if thisLoc['fullitemname'] == "Magma Chamber":
-            thisLoc['inlogic'] = depthsL and (((Varia in loadout) and (Charge in loadout)) or ((MetroidSuit in loadout) and energyCount > 6))
-        if thisLoc['fullitemname'] == "Equipment Locker":
-            thisLoc['inlogic'] = pirateLab
-        if thisLoc['fullitemname'] == "Antelier": # spelled "Antilier" in subversion 1.1
-            thisLoc['inlogic'] = pirateLab and underwater
-        if thisLoc['fullitemname'] == "Weapon Research":
-            thisLoc['inlogic'] = vulnar and canUseBombs and (Wave in loadout) and underwater and ((DarkVisor in loadout) or (pirateLab and (canUsePB or (Screw in loadout))))
-        if thisLoc['fullitemname'] == "Crocomire's Lair":
-            thisLoc['inlogic'] = eastLomyr and (Super in loadout) and (SpeedBooster in loadout)
+    # onAndOff = ON varia ice PB grapple OFF Hyper charge
+    late_spaceport = (
+        (spaceDrop in loadout) and
+        geothermal and
+        (Grapple in loadout) and
+        (Screw in loadout) and
+        (MetroidSuit in loadout)
+    )
```

Comment:
I saw this group repeated in a bunch of places, and I called it late_spaceport as in when you return to the spaceport.
Did I interpret that correctly?
Looking at it more closely now, I'm thinking maybe I didn't interpret it correctly, because Screw isn't required to get back to spaceport, and MetroidSuit isn't required if you turn off the power...

What is this group that I saw repeated in a bunch of places?

It was on all of these in expert:
"Weapon Locker", "Aft Battery", "Gantry", "Docking Port 4", "Ready Room", "Torpedo Bay", "Extract Storage"

"Extract Storage" is the vanilla MetroidSuit location, right?